recipe/patch_yaml/numpy2_known_lower_bound.yaml

```diff
@@ -0,0 +1,405 @@
+# packages from https://github.com/numpy/numpy/issues/26191
+# where we know the lowest version that supports numpy 2.0
+if:
+  name: astropy
+  version_lt: 6.1.0
+  has_depends: python >=3.9*
+  not_has_depends: numpy *,<2.0a0
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+---
+if:
+  name: awkward
+  version_lt: 2.6.3
+  has_depends: python >=3.9*
+  not_has_depends: numpy *,<2.0a0
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+---
+if:
+  name: bokeh
+  version_lt: 3.4.1
+  has_depends: python >=3.9*
+  not_has_depends: numpy *,<2.0a0
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+---
+if:
+  # legacy name for libboost-python
+  name: boost
+  version_lt: 1.86.0
+  has_depends: python >=3.9*
+  not_has_depends: numpy *,<2.0a0
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+---
+if:
+  name: cartopy
+  version_lt: 0.23.0
+  has_depends: python >=3.9*
+  not_has_depends: numpy *,<2.0a0
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+---
+if:
+  name: contourpy
+  version_lt: 1.2.1
+  has_depends: python >=3.9*
+  not_has_depends: numpy *,<2.0a0
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
+---
+if:
+  name: cupy
+  version_lt: 14.0.0
+  has_depends: python >=3.9*
+  not_has_depends: numpy *,<2.0a0
+then:
+  - tighten_depends:
+      name: numpy
+      upper_bound: "2.0"
```

Comment:
Just a note, in the `cupy` case this only recently became `noarch` (as part of the `13.0.0` upgrade). So the `has_depends: python >=3.9*` only applies to recent `cupy` packages. However the ones without the `numpy` pin preceded the move of `cupy` to `noarch`. So in that case we want to drop this constraint

This is a bit of a niche case most likely. Though wanted to mention here for completeness in case it comes elsewhere

Fixing in PR: https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/873